sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java

```diff
@@ -339,41 +248,34 @@ public void populateDisplayData(DisplayData.Builder builder) {
 
     if (mode == Mode.FILEPATTERN) {
       long startTime = System.currentTimeMillis();
-      List<ListenableFuture<List<? extends FileBasedSource<T>>>> futures = new ArrayList<>();
-
-      ListeningExecutorService service =
-          MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(THREAD_POOL_SIZE));
-      try {
-        checkState(fileOrPatternSpec.isAccessible(),
-                   "Bundle splitting should only happen at execution time.");
-        Collection<String> expandedFiles =
-            FileBasedSource.expandFilePattern(fileOrPatternSpec.get());
-        checkArgument(!expandedFiles.isEmpty(),
-            "Unable to find any files matching %s", fileOrPatternSpec.get());
-        for (final String file : expandedFiles) {
-          futures.add(createFutureForFileSplit(file, desiredBundleSizeBytes, options, service));
-        }
-        List<? extends FileBasedSource<T>> splitResults =
-            ImmutableList.copyOf(Iterables.concat(Futures.allAsList(futures).get()));
-        LOG.info(
-            "Splitting filepattern {} into bundles of size {} took {} ms "
-                + "and produced {} files and {} bundles",
-            fileOrPatternSpec,
-            desiredBundleSizeBytes,
-            System.currentTimeMillis() - startTime,
-            expandedFiles.size(),
-            splitResults.size());
-        return splitResults;
-      } finally {
-        service.shutdown();
+      checkState(fileOrPatternSpec.isAccessible(),
+                 "Bundle splitting should only happen at execution time.");
+      List<Metadata> expandedFiles = FileBasedSource.expandFilePattern(fileOrPatternSpec.get());
```

Comment:
I suppose this can now be synchronous cause we don't need to size the files, right?